connectivity/check/check.go

```diff
@@ -1579,6 +1593,21 @@ func (k *K8sConnectivityCheck) validateDeployment(ctx context.Context) error {
 		return err
 	}
 
+	k.ciliumPods = map[string]PodContext{}
+	for _, client := range k.clients.clients() {
+		ciliumPods, err := client.ListPods(ctx, k.ciliumNamespace, metav1.ListOptions{LabelSelector: "k8s-app=cilium"})
+		if err != nil {
+			return fmt.Errorf("unable to list Cilium pods: %s", err)
+		}
+		for _, ciliumPod := range ciliumPods.Items {
+			// TODO: Can Cilium pod names collide across clusters?
```

Comment:
DaemonSet Pod names overlapping across clusters is not unlikely, since they only contain 5 bytes of entropy (and only `a-z0-9`). I'll pick this up during the rebase, there have already been some changes to make resource naming more consistent.